sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/CancellationTokenSourceExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+
+namespace Azure.Messaging.ServiceBus
+{
+    internal static class CancellationTokenSourceExtensions
+    {
+        public static void CancelAfter(this CancellationTokenSource cancellationTokenSource,
+            ServiceBusReceivedMessage receivedMessage)
+        {
+            if (cancellationTokenSource == null || receivedMessage.LockedUntil == default)
+            {
+                return;
+            }
+
+            var delay = receivedMessage.LockedUntil.Subtract(DateTimeOffset.UtcNow);
```

Comment:
Do we have anything from the service available in the annotated message that would allow to check against instead of `DateTimeOffset.UtcNow` to avoid clock drift?

Reply:
Unfortunately, not to my knowledge.  There's no server time communicated nor way to query it.

Reply:
Ok then I guess this has to suffice. Should I add a remark to the token talking about potential drift scenarios?